FKMS: max refresh rate and blocking 1366x768

6by9 opened the pull request.

6by9 committed:
drm/vc4: Make FKMS max refresh rate a module parameter

Some people want to use the high refresh rate modes for 1080p100
and 1080p120, but they're currently filtered out as generally
they don't add anything.

Make the filter threshold a module parameter so that it can be
adjusted.

https://github.com/raspberrypi/linux/issues/3677

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

popcornmix commented on drivers/gpu/drm/vc4/vc4_firmware_kms.c:
```diff
@@ -1097,6 +1099,17 @@ vc4_crtc_mode_valid(struct drm_crtc *crtc, const struct drm_display_mode *mode)
+	     ((mode->hsync_start - mode->hdisplay) & 1) ||  /* front porch */
+	     ((mode->hsync_end - mode->hsync_start) & 1) || /* sync pulse */
+	     ((mode->htotal - mode->hsync_end) & 1)))       /* back porch */
+		return MODE_H_ILLEGAL;
```
`(A|B|C|D)&1` should generate better code, but it's arguable which is clearer.

6by9 replied:
I've just done exactly that in the firmware patch. 

It's not a path taken that often, so optimising a few ARM instructions out seems a little excessive. I'll let @pelwell make the call.

pelwell replied:
You could compromise and just change the `||`s to `|`s, which might well generate equally good code.

6by9 replied:
That sounds less readable.
I've gone for the ```(A|B|C|D)&1```

pelwell replied:
Opinions, etc.

pelwell commented:
Can you split the DTS change into a separate commit? DT files are regularly squashed down to a single downstream DT commit, and having to split this just more work for @popcornmix.

6by9 committed:
dt: Use rpi-firmware-kms-2711 on 2711 platforms

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 committed:
drm/vc4: FKMS Block modes with odd horizontal timing values on Pi4

Pi4 HDMI pipeline is 2 pixels/clock and can not produce timings
that have odd values for active pixels, front porch, sync width,
or back porch.
Detect these modes and block them within fkms.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 committed:
drm/vc4: FKMS: Put includes in alphabetical order, and use <> instead of ""

Reorder the includes, and use the system include paths rather than
local ones

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 commented:
DT split out, using ```(A|B|C|D)&1```, and cleaned up the includes now as I found out I'd included <linux/module.h> that was already included as "linux/module.h".

pelwell commented on drivers/gpu/drm/vc4/vc4_firmware_kms.c:
```diff
@@ -1097,6 +1098,17 @@ vc4_crtc_mode_valid(struct drm_crtc *crtc, const struct drm_display_mode *mode)
+	if (fkms->bcm2711 &&
+	    (vc4_crtc->display_number == 2 || vc4_crtc->display_number == 7) &&
+	    ((mode->hdisplay |				/* active */
+	     (mode->hsync_start - mode->hdisplay) |	/* front porch */
```
Sorry to be a stickler, but this line and the following few would benefit from an extra preceding space - everything else is so beautifully indented.

6by9 replied:
You can go off a person you know!

pelwell replied:
Only if you like them in the first place.
